File: programmers/level1/콜라츠_추측.java
package programmers;

public class 콜라츠_추측 {
	static
	class Solution {
		  public int solution(int num) {
		      long temp = num;
		      for(int i = 0; i < 500; ++i) {
		           if(temp==1){
		                  return i;
		              }
		          if(temp % 2 == 0) {
		              temp /= 2;
		             
		          }else {
		              temp = temp * 3 + 1;
		          }
		      }
		      return -1;
		  }
		}
	public static void main(String[] args) {
		Solution sol = new Solution();
		int n = 6;
		System.out.println(sol.solution(n));
	}
}

File: programmers/python/level1/콜라츠_추측.py
def solution(num):
    answer = 0
    while answer < 500:
        if num == 1:
            break
        if num % 2 == 0:
            num /= 2
        else:
            num = num * 3 + 1
        answer += 1
    if answer == 500:
        answer = -1
    return answer